src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,33 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's contact in the address book.
+ */
+public class Contact {
+
+    public String value;
+    protected boolean isPrivate;
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
+                && this.value.equals(((Contact) other).value)); // state check
```

Comment:
Mentioned in class that when two Contact objects have the same value, but of a different child class, they can be "equal" under with this method. The equals comparison should be done in the child classes.